broadinstitute/single_cell_portal_core
# Enable canvas recording for Sentry replays (SCP-5826)

#### BACKGROUND & CHANGES
This updates the Sentry SDK to `7.119.1` to enable recording of canvas & WebGL DOM components during Session Replays and also address an open [Dependabot alert](https://github.com/broadinstitute/single_cell_portal_core/security/dependabot/159).  This will allow us much greater insight into user behavior when triaging front-end errors.

Demo of new replays (fast-forward to ~6:15 to see the new plots): https://broad-institute.sentry.io/replays/d6ee1185376e4b70832e94a965801330/?environment=development&project=1424198&query=&referrer=%2Freplays%2F&statsPeriod=14d&yAxis=count%28%29

#### MANUAL TESTING
1. Boot as normal
2. In your IDE, open `app/javascript/lib/sentry-logging.js` and in `getIsSuppressedEnv()` to add `return false` at line 71
3. On line 122, set `replaysOnErrorSampleRate` to `1.0`
4. Back in a browser, load any study with fully initialized plots
5. Query for one gene, and then once the plot has loaded use the `Return to cluster view` button immediately to the right of the gene search bar:
![Screenshot 2024-10-08 at 11 12 32 AM](https://github.com/user-attachments/assets/476cdd74-f4cc-4bd6-a2b3-09d3e2baa48d)
6. This will throw an error that is reported to Sentry (you can confirm this in the JS console & network tab) that should also now include a replay
7. Go to [Sentry Replays](https://broad-institute.sentry.io/replays/?environment=development&project=1424198&statsPeriod=14d), make sure you have the `development` environment selected, and then find your replay, confirming you see the content of scatter plots
